### PR TITLE
Fixed #34344 -- Allowed PostGIS DatabaseWrapper subclasses to customize (features/introspection/ops)_class classes.

### PR DESCRIPTION
**Pull request by davidjayb**
This allows for extensions to override `*_class` properties when extending the backend.

I came across an issue when trying to add PostGIS support to the django-multitenant library:

https://github.com/citusdata/django-multitenant/pull/150/files#diff-0d077af5d4e86f31f478785458ea6a03768a15aea077f49b9a79ffca8b0fe4beR37

Tracking down the issue took a bit of time as it wasn't apparent that the PostGIS backend was explicitly setting the features property to the DatabaseFeatures.

This maintains backwards compatibility for the fix to https://code.djangoproject.com/ticket/16969

### CODE REVIEW

File: django/contrib/gis/db/backends/postgis/base.py [hunk 1]
```diff
@@ -89,11 +92,19 @@ class DatabaseWrapper(PsycopgDatabaseWrapper):
     }
 
     def __init__(self, *args, **kwargs):
+        if kwargs.get("alias", "") == NO_DB_ALIAS:
+            # Reset the *_class variables to the postgresql implementations
```
Comment (felixxm) [inline django/contrib/gis/db/backends/postgis/base.py:96]:
```suggestion
            # Don't initialize PostGIS-specific stuff for non-db connections.
```

**Comment by felixxm** [comment]
I've prepared a regression tests for ticket-16969, see #16582.

**Comment by felixxm** [comment]
@davidjayb Thanks :+1: Welcome aboard :boat: 

I moved import to the top.

**Comment by davidjayb** [comment]
Thank you @felixxm

**Pull request merged**